### PR TITLE
Fix Blackboard and Autolab

### PR DESCRIPTION
**Pull request by apjaffe**
This branch updates the code to retrieve grades from Blackboard and Autolab.

### CODE REVIEW

File: auth.py [hunk 1]
```diff
@@ -16,12 +16,15 @@ def authenticate(url):
     s = requests.Session()
 
     # 1. Initiate sequence by querying the protected asset
-    s.get(url)
+    if("https://autolab.cs.cmu.edu/" in url):
+    	s.get(url, verify=False)
```
Comment (willcrichton) [inline auth.py]: Why does verify=False have to be passed to Autolab?

Comment (apjaffe) [inline auth.py]: Autolab seems to have their certificates misconfigured, if I leave out that argument I get an error: "SSL3_GET_SERVER_CERTIFICATE:certificate verify failed"

Comment (willcrichton) [inline auth.py]: Lolwut. Poor Autolab. Is there any way that you can work around this outside of the authentication function, i.e. authenticate to an http:// URL instead? I'm not a fan of having URL-specific hacks inside the auth function wherever possible. 

Alternatively, you could add an additional optional parameter dictionary to the authenticate function and then pass that in to the session call.

Comment (apjaffe) [inline auth.py]: Okay, I added the parameter dictionary.

**Pull request merged**